pkg/graph/resolvers/model_plan.go

```diff
@@ -13,13 +13,10 @@ import (
 // TODO Revist this function, as we probably want to add all of these DB entries inthe scope of a single SQL transaction
 // so that we can roll back if there is an error with any of these calls.
 func ModelPlanCreate(logger *zap.Logger, modelName string, store *storage.Store, principalInfo *models.UserInfo, principal authentication.Principal) (*models.ModelPlan, error) {
-	plan := &models.ModelPlan{
-		ModelName: modelName,
-		Status:    models.ModelStatusPlanDraft,
-		BaseStruct: models.BaseStruct{
-			CreatedBy: principalInfo.EuaUserID,
-		},
-	}
+
+	plan := models.NewModelPlan(principal.ID())
+	plan.ModelName = modelName
+	plan.Status = models.ModelStatusPlanDraft
```

Comment:
I think I like this approach. It's a bit of a departure from our current approach, but I think this also provides a lot more value overall to the idea of using constructors.  This seems to be a specific use case when we are actually preparing to create a new record for the DB, rather than instantiating the record that is already in the DB. (If I'm selecting the record from the DB, I can just make a model plan to receive the values from the read like rePlan :-= models.ModelPlan{})

The only thing I'm thinking of that is slightly different is that created_dts is still set on the DB level. I think it still is valid to say that we would be setting all values that are needed to do an insert operation into the database however.

Reply:
Yeah. We could just as easily `time.Now()` and make it so the DB is responsible for nothing, and all values need to be provided in the struct.

Could also say the same for other default values like the UUID, but I think I'd prefer the application handle the values going into the DB in most cases.
```sql
CREATE TABLE model_plan (
  id UUID PRIMARY KEY NOT NULL DEFAULT uuid_generate_v4(),
  model_name TEXT NOT NULL
)
```

Reply:
Hmm yeah, that is interesting callout too @ClayBenson94. In some ways the thought of the DB generating UUID makes sense, because it knows if that specific UUID is already taken for another row. I think that's a bit of a moot point though since UUID is designed in a manner to make conflicts very unlikely. I'd be inclined not to change the paradigm for date in this ticket ( or UUID), but I think it could be something we revisit and unify in the code base.